### PR TITLE
Block controls: Animate block controls as they appear

### PR DESCRIPTION
**Pull request by youknowriad**
Small PR adding an animation to block controls to match the prototypes

refs #15

### CODE REVIEW

**Comment by jasmussen** [comment]
This looks and works great! Solid work!

When I move a block upwards, the controls stay in place. When I move a block downwards, the animation to make the controls appear plays again. The desired behavior would be no animation of the controls when moving blocks.

**Comment by youknowriad** [comment]
> When I move a block downwards, the animation to make the controls appear plays again.

I'm afraid we can not do much about this (at least not easily). I suspect this is caused by the way React handles component reordering. We'll need to leverage some React Animation library to perform this. Maybe https://github.com/reactjs/react-transition-group or https://github.com/chenglou/react-motion. @aduth any feedback on these libraries or Animations in React in general?

**Comment by jasmussen** [comment]
Going to try a thing with animation-play-state: paused; real quick.

**Comment by jasmussen** [comment]
So, the play-state idea didn't work, but here's another that might. 

```
.editor-visual-editor__block-controls {
	@include animate_fade;
	position: absolute;
	bottom: 100%;
	margin-bottom: -4px;
	left: 43px;
	display: flex;

	.is-moving & {
		animation-delay: -.2s;
	}
}
```

That is —  the negative animation delay basically starts the animation on its last frame. But we need a CSS class to indicate that the block is being moved for this to work. Do you think this makes sense/is worth it?

If not, then it's not a blocker for merge, definitely something we can live with.

**Comment by youknowriad** [comment]
> But we need a CSS class to indicate that the block is being moved for this to work. Do you think this makes sense/is worth it?

Adding these "animation" classes is exactly what  https://github.com/reactjs/react-transition-group does, it's a helper to avoid adding complexity to the code to handle animations.

**Review by jasmussen — APPROVED** [review]
Maybe worth revisiting some animation tools to enhance states, but not a blocker for merge.

File: webpack.config.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const config = {
 							loader: 'sass-loader',
 							query: {
 								includePaths: [ 'editor/assets/stylesheets' ],
-								data: '@import "variables"; @import "mixins";',
+								data: '@import "variables"; @import "mixins"; @import "animations";',
```
Comment (aduth) [inline webpack.config.js]: We might consider creating a `common.scss` to import these files so we don't have to manage the Webpack configuration directly.

File: editor/modes/visual-editor/block.js [hunk 2]
```diff
@@ -108,17 +108,19 @@ class VisualEditorBlock extends wp.element.Component {
...
-					{ isSelected && ! isTyping && settings.controls ? (
-						<Toolbar
-							controls={ settings.controls.map( ( control ) => ( {
-								...control,
-								onClick: () => control.onClick( block.attributes, this.setAttributes ),
-								isActive: () => control.isActive( block.attributes )
-							} ) ) } />
-					) : null }
-				</div>
+				{ isSelected && ! isTyping &&
```
Comment (aduth) [inline editor/modes/visual-editor/block.js]: Nice idea to move the condition outside the common parent 👍

**Pull request merged**